lib/Mojo/WebSocketProxy/RequestLogger.pm

```diff
@@ -0,0 +1,30 @@
+use Object::Pad;
+
+class Mojo::WebSocketProxy::RequestLogger;
+
+use strict;
+use warnings;
+use Log::Any qw($log);
+use UUID::Tiny;
+
+field $context;
+BUILD{
+    $context->{correlation_id} =  UUID::Tiny::create_UUID_as_string(UUID::Tiny::UUID_V4);
+}
+
+# Method to log messages with various log levels
+method log_message($level, $message, @params) {
+    if($log->adapter->can('set_context')) {
+        $log->adapter->set_context($context);
+    }
+    $log->$level($message, @params);
+    if($log->adapter->can('clear_context')) {
+        $log->adapter->clear_context;
+    }
```

Comment:
why we need set_context and clear_context ? 
why not 
```
local $log->context->{$_} = $context->{$_} for keys %$context;
```

Reply:
we are using the Deriv adapter function here for setting the context in log object and then clearing after its usage

Reply:
I don't think that's a good  idea to clear_context. That will make log mess. Suppose you want to print a context all time:
```
$log->context->{hello} = "world";
```
and after that you want to set context:
```
$log->adapter->set_context({foo => "bar"});
$log->info("blahblah");
$log->adapter->clear_context;
```
then after that the context 'hello' wlll be discarded.
that's not a good idea.
And, I guess we shouldn't call $log->adapter in most cases.

what we should do is:

```
for my $key (keys %$context){
    local $log->context->{$key} = $context->{$key};
}
$log->$level($message, @params);

```

Reply:
There are two contexts @chylli-deriv:
`$log->context`: is the one the module itself supports. it's global and limited as per the current implementation of the module and doing changes on it is a bit tricky.
`$log->adapter->context` is the one introduced by @aisha-deriv and we have a bit of control over it only in the `DERIV` adapter without affecting the complete module.
We can probably also provide support for the global context inside the new one.

Reply:
@aisha-deriv we can introduce an accessor method to the adapter and expose the context:
```
sub context {my $self = shift; return $self->{context}};
```
Then we can happily take advantage of the `local` keyword.

Another way around the local is to define a setter and clearer for the RequestLogger in which it sets and clears its own context out of the adapter context.

Reply:
@chylli-deriv  as @hani said we are using adapter context because of the format we have applied to it and its not global
@hani-deriv adding this accessor will not lead issues with random context values ? with the clearing context we are always sure that correct context is put in message

Reply:
We extend the functionality only, and we keep the ability to completely clear the context.
Depending on the usage could be random or *global* values.
For the `RequestLogger` if it sets its context and clears it. no chance for these random values, unless something went wrong.

Reply:
There is always a room for enhancement we can cater the request for localize the context as @chylli-deriv mentioned later to the adapter. For now we strictly want to keep the context availability similar for all messages so the scope is defined to be per request for log messages.